### PR TITLE
ARROW-12370: [R] Bindings for power kernel

### PR DESCRIPTION
**Pull request by pachadotdev**

### CODE REVIEW

**Review by nealrichardson — COMMENTED** [review]
Looks good. Should we also add a test of calling it through `dplyr`, just to be sure/explicit? (The answer may be no.)

Should we test with other data types as well? Not that we need R tests for this, but just to confirm that the C++ implementation is doing what it should with types other than float64?

**Comment by pachamaltese** [comment]
> Looks good. Should we also add a test of calling it through `dplyr`, just to be sure/explicit? (The answer may be no.)
> 
> Should we test with other data types as well? Not that we need R tests for this, but just to confirm that the C++ implementation is doing what it should with types other than float64?

sure ! adding now

**Comment by pachamaltese** [comment]
> this

the only part I don't understand is a dplyr test here

File: r/tests/testthat/test-dplyr-filter.R [hunk 1]
```diff
@@ -284,7 +300,7 @@ test_that("filter environment scope", {
   # Also for functions
   # 'could not find function "isEqualTo"' because we haven't defined it yet
   expect_dplyr_error(input %>% filter(isEqualTo(int, 4)), tbl)
-  
+
```
Comment (nealrichardson) [inline r/tests/testthat/test-dplyr-filter.R:303]:
```suggestion
```

File: r/tests/testthat/test-dplyr-filter.R [hunk 2]
```diff
@@ -399,5 +415,5 @@ test_that("filter() with .data pronoun", {
       collect(),
     tbl
   )
-  
+
```
Comment (nealrichardson) [inline r/tests/testthat/test-dplyr-filter.R:418]:
```suggestion
```

**Review by nealrichardson — APPROVED** [review]
I'll trim this whitespace and then merge, thanks!

**Pull request closed without merging**